processors/base_processor.rb

```diff
@@ -12,12 +14,48 @@ def self.call(page: 1)
   end
 
   def call
-    raise NotImplementedError, 'Method #call has to be implemented!'
+    visit_page
+    parse_item_links
+    visit_and_parse_items
+
+    return browser.exit_browser unless page_parser.next_page_exists?
+
+    @page += 1
```

Comment:
I think this string is useless, because you check whether next page exist

Reply:
I use it as page number at the end of URL. Page knowledge is required